### DISCUSSION TITLE
Organization policy enforment not applying

### DISCUSSION THREAD

**Discussion by oshen007** [dani-garcia/vaultwarden#4955] [Q&A]
With the latest release of vaultwarden the organization policies are not being enforced. I tested the setting the **Remove individual vault** and **Remove Send** are not being applied. The user under that organization still can create sends, their personal vault still shows and 2FA is not enforced. It seems like the password policies are the only thing working. See screenshots for more details: 

![image](https://github.com/user-attachments/assets/b395c63e-b599-4e8c-a5c2-cebf045d48f4)

![image](https://github.com/user-attachments/assets/0aa5a253-44bf-4e8d-be02-86114694021a)

![image](https://github.com/user-attachments/assets/b9edd02f-ea0b-477d-a5d3-beade21fe950)
 
As shown the LA user is a member of the org as a User and the policies are not being applied.
Does anyone have an idea what i'm doing wrong or how to fix it?

**Comment by stefan0xC** [comment]
Have you tried creating a send or an item in the personal vault after activating the respective policy? Because you shouldn't be able to. (The sends are still visible because you should be able to delete existing sends, same with existing items in the individual vault.)

Also the policies will not be enforced for owners/admins of an organization. 
Cf. https://bitwarden.com/help/policies/#remove-individual-vault

**Reply by oshen007** [reply]
The send and the items were all created after the user was added to the organization. For more context the user was created directly through the invite members option under the organization.

**Reply by stefan0xC** [reply]
@oshen007 I can't reproduce that because when I create a new organization, activate those policies and invite a new user, I get an error message as the new user when I click save for both items in the personal vault as well as sends.

```
[2024-09-15 00:35:45.004][request][INFO] POST /api/ciphers
[2024-09-15 00:35:45.009][vaultwarden::api::core::ciphers][ERROR] Due to an Enterprise Policy, you are restricted from saving items to your personal vault.
[2024-09-15 00:35:45.009][response][INFO] (post_ciphers) POST /api/ciphers => 400 Bad Request
[...]
[2024-09-15 00:36:04.552][request][INFO] POST /api/sends/file/v2
[2024-09-15 00:36:04.558][vaultwarden::api::core::sends][ERROR] Due to an Enterprise Policy, you are only able to delete an existing Send.
[2024-09-15 00:36:04.559][response][INFO] (post_send_file_v2) POST /api/sends/file/v2 => 400 Bad Request
```

**Reply by oshen007** [reply]
When I look into the user's profile it says I'm not part of any organization even though i clearly accepted to join the organization through the link and created the user account. When I click on the users **my vault** the url shows **organizationId=unassigned** 

![image](https://github.com/user-attachments/assets/add1ea3e-7f7d-44ee-b811-023dfac6e80b)

**Reply by stefan0xC** [reply]
Are you sure the user accepted the organization invitation?

**Reply by stefan0xC** [reply]
@oshen007 you require users to setup 2FA before joining an organization, so most likely the user in question has not been added because of that.

**Reply by oshen007** [reply]
Yes, I personally clicked Join the organization button that was sent to the email and created the user.

**Reply by oshen007** [reply]
@stefan0xC I will disable and delete the user and try again and get back with you

**Reply by oshen007** [reply]
@stefan0xC I believe I found what my error was. I was creating the user through the Join the organization link and it was rejecting the user from joining due to the 2FA not being set as it was a newly created user. I had to enable 2FA and then click on the join the organization link again and it allowed me to join.

It would be helpful if the system prompted users to set up 2FA immediately after account creation and before allowing them to login and join an organization. This would streamline the process and improve the user experience.

**Reply by stefan0xC** [reply]
> It would be helpful if the system prompted users to set up 2FA immediately after account creation and before allowing them to login and join an organization. This would streamline the process and improve the user experience.

Yeah, I think the idea is that it should enable email as 2FA provider automatically in these cases. Seems like this doesn't work as expected. update: should be fixed via #4956

**Reply by Gerardv514** [reply]
What happens if a VW instance doesn’t have an smtp server configured?

**Reply by stefan0xC** [reply]
Then email as 2fa is not enabled
https://github.com/dani-garcia/vaultwarden/blob/6ceed9284d16ba11cae4515d6ab666225725cb99/src/config.rs#L710
